modules/fbs/fbs.module

```diff
@@ -571,12 +571,23 @@ function fbs_form_profile2_phone_validate($element, $form_state) {
   $phone = $form_state['input']['profile_provider_fbs']['field_fbs_phone']['und'][0]['value'];
 
   if (!empty($phone)) {
-    // Preg match for danish phonenumbers
-    // Pass:
-    // (+45) 35 35 35 35 ||| +45 35 35 35 35 ||| 35 35 35 35 ||| 35353535
-    // Fail:
-    // (45)35353535 ||| 4535353535.
-    $re = '/^((\(?\+45\)?)?)(\s?\d{2}\s?\d{2}\s?\d{2}\s?\d{2})$/';
+    // Preg match for phonenumbers
+    // Pass (danish phonenumbers):
+    // (+45) 35 35 35 35 ||| +45 35 35 35 35 ||| 35 35 35 35 |||
+    // (+45)35353535 ||| +4535353535 ||| 35353535
+    // Pass (german phonenumbers):
+    // (+49) 089/12345678 ||| (+49) 089-12345678 ||| +49 089/12345678 ||| +49 089-12345678
+    // +49089/12345678 ||| +49089-12345678 ||| (+49)(089) 12345678
+    // Allows for countrycode with, and without parenthesis.
+    // Allows for spaces between numberpairs in number
+    // Allows for / and - in number (used in german phonenumbers).
+    // First matching group: ([\(]{0,1}[\+][0-9]{1,3}[ \.\-]?[\)]{0,1})
+    // Matches countrycode (with and without parenthesis)
+    // Second matching group: ([\(]{1}[0-9]{1,6}[\)])
+    // Matches local area codes with upto 5 digits eg. german (089)
+    // Third matching group: ([0-9 \.\-\/]{3,20})((x|ext|extension)[ ]?[0-9]{1,4})
+    // Matches main phonenumber - allowes for special characters . - / in number.
+    $re = '/^([\(]{0,1}[\+][0-9]{1,3}[ \.\-]?[\)]{0,1})?([\(]{1}[0-9]{1,6}[\)])?([0-9 \.\-\/]{3,20})((x|ext|extension)[ ]?[0-9]{1,4})?$/';
```

Comment:
Why is the whole extension part relevant?

Reply:
This part is added to validate all phonenumbers, also with local extensions - but maybe could/should be left out in this case? A number like this would be validated: 415-555-4242 ext99. Read more about extensions here: https://getvoip.com/library/difference-between-a-phone-number-an-extension-and-a-line/